Source/WTF/Configurations/module.modulemap

```diff
@@ -0,0 +1,62 @@
+module wtf {
+    // AutodrainedPool must never be parsed in an objective-C
+    // context.
+    explicit module AutodrainedPool {
+        requires cplusplus11, !objc
+        header "AutodrainedPool.h"
+        export *
+    }
+    // This header depends on surrounding preprocessor context.
+    // Works only on some platforms.
+    explicit module MetadataSPI {
+        requires cplusplus11
+        textual header "spi/mac/MetadataSPI.h"
+        export *
+    }
+    // The following WTF headers can be used in C contexts as well as C++
+    // so we put them in a separate submodule. We'll call it Platform
+    // since the majority of the files are Platform-related.
+    // (Some are interdependent so we couldn't split them into
+    // totally standalone submodules even if we wanted to.)
+    explicit module Platform {
+        header "Assertions.h"
+        header "Compiler.h"
+        header "ExportMacros.h"
+        header "spi/cocoa/IOSurfaceSPI.h"
+        header "SwiftBridging.h"
+        header "Platform.h"
+        header "PlatformCallingConventions.h"
+        header "PlatformCPU.h"
+        header "PlatformEnable.h"
+        header "PlatformEnableCocoa.h"
+        header "PlatformHave.h"
+        header "PlatformLegacy.h"
+        header "PlatformOS.h"
+        header "PlatformUse.h"
+        export *
+    }
+    // Swiftc cannot currently parse various bits of WTF so we have to
```

Comment:
I don't really understand this. If Swift doesn't import the header correctly then shouldn't this be `requires !swift`?

Reply:
Swift _should_ be able to parse these but can't due to temporary bugs, so it feels wrong to declare that these are intentionally `!swift`. I could, but there's no real need. So long as no Swift code actually uses these headers, it doesn't attempt to import them.

I don't really mind here.

Reply:
Ok. The comment's kind of a moot point if all the other headers get their own explicit submodule but 👍